### PR TITLE
Read Stage 0 DICE data in Stage 1

### PR DESCRIPTION
**Pull request by conradgrobler**
For now we just read the DICE data from the memory addres that Stage 0 sent. We don't yet do anything useful with it apart from logging the magic number.

Converting the DICE data to a proto and adding a measurement for the next stage will be done as follow-up PRs..

### CODE REVIEW

File: oak_containers_stage1/src/dice.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        .expect("couldn't open DICE memory device for reading");
+    dice_reader
+        .seek(std::io::SeekFrom::Start(start.as_u64()))
+        .expect("couldn't seek to DICE offset");
+    dice_reader
+        .read_exact(buffer)
+        .expect("couldn't read DICE data");
+
+    if result.magic != STAGE0_MAGIC {
+        anyhow::bail!("invalid DICE data");
```
Comment (andrisaar) [inline oak_containers_stage1/src/dice.rs:72]: the expect's (which panic) and bail!'s (which return Err) are a bit inconsistent in this function in my opinion. How did you decide when to panic and when to return an error?

File: oak_containers_stage1/src/main.rs [hunk 2]
```diff
@@ -146,3 +171,17 @@ async fn main() -> Result<(), Box<dyn Error>> {
 
     image::switch(&args.init).context("error switching to the system image")?
 }
+
+/// Tries to parse a string slice as an address.
+///
+/// Assumes that the address is a hexadecimal representation if it starts with "0x", or decimal
+/// otherwise.
+fn try_parse_phys_addr(arg: &str) -> anyhow::Result<PhysAddr> {
```
Comment (andrisaar) [inline oak_containers_stage1/src/main.rs:179]: I'm surprised there's not an easier way to do this :(

**Review by andrisaar — APPROVED** [review]

File: oak_containers_stage1/src/dice.rs (hunk 1 above)
Comment (conradgrobler) [reply oak_containers_stage1/src/dice.rs:72]: Good point. I copied the code with the expects from an earlier prototype. Updated it now with ``context` and `?` to be more consistent.

File: oak_containers_stage1/src/main.rs (hunk 2 above)
Comment (conradgrobler) [reply oak_containers_stage1/src/main.rs:179]: Yes, I couldn't find a better way.

**Review by conradgrobler — COMMENTED** [review]
Thanks

**Pull request merged**